cpp/daal/src/data_management/roc_auc_score.cpp

```diff
@@ -0,0 +1,165 @@
+/** file roc_auc_score.cpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "data_management/data/internal/roc_auc_score.h"
+#include "data_management/data/numeric_table.h"
+#include "services/daal_defines.h"
+#include "src/algorithms/service_kernel_math.h"
+#include "src/algorithms/service_sort.h"
+#include "src/externals/service_dispatch.h"
+#include "src/externals/service_memory.h"
+#include "src/services/service_data_utils.h"
+#include "src/threading/threading.h"
+
+template <typename FPType>
+void sort(IdxValType<FPType> * beginPtr, IdxValType<FPType> * endPtr)
+{}
+
+template <>
+void sort<float>(IdxValType<float> * beginPtr, IdxValType<float> * endPtr)
+{
+    daal::parallel_sort_pair_fp32_uint64(beginPtr, endPtr);
+}
+
+template <>
+void sort<double>(IdxValType<double> * beginPtr, IdxValType<double> * endPtr)
+{
+    daal::parallel_sort_pair_fp64_uint64(beginPtr, endPtr);
+}
+
+namespace daal
+{
+namespace data_management
+{
+namespace internal
+{
+template <typename FPType, daal::CpuType cpu>
+services::Status calculateRankDataImpl(FPType * predictedRank, const NumericTablePtr & testPrediction, const size_t & nElements)
+{
+    services::Status s;
+    ReadRows<FPType, cpu> testPredictionBlock(testPrediction.get(), 0, 1);
+    const FPType * const testPredictionPtr = testPredictionBlock.get();
+    DAAL_CHECK_BLOCK_STATUS(testPredictionBlock);
+
+    TArray<IdxValType<FPType>, cpu> predict(nElements);
```

Comment:
Use `TArrayScalable` instead `TArray` for parallel regions
```suggestion
    TArrayScalable<IdxValType<FPType>, cpu> predict(nElements);
```